fearless/Modules/NewWallet/ChainAccount/ChainAccountInteractor.swift

```diff
@@ -7,52 +7,36 @@ import SoraKeystore
 final class ChainAccountInteractor {
     weak var presenter: ChainAccountInteractorOutputProtocol?
     var chainAsset: ChainAsset
-    var availableChainAssets: [ChainAsset] = []
+    var availableChainAssets: [ChainAsset]
 
     private var wallet: MetaAccountModel
     private let operationManager: OperationManagerProtocol
     private let eventCenter: EventCenterProtocol
     private let repository: AnyDataProviderRepository<MetaAccountModel>
     private let availableExportOptionsProvider: AvailableExportOptionsProviderProtocol
-    private let chainAssetFetching: ChainAssetFetchingProtocol
 
     init(
         wallet: MetaAccountModel,
         chainAsset: ChainAsset,
+        availableChainAssets: [ChainAsset],
         operationManager: OperationManagerProtocol,
         eventCenter: EventCenterProtocol,
         repository: AnyDataProviderRepository<MetaAccountModel>,
-        availableExportOptionsProvider: AvailableExportOptionsProviderProtocol,
-        chainAssetFetching: ChainAssetFetchingProtocol
```

Comment:
module should be independence, please leave chainAssetFetching and pass ChainEcosystem to filter fetched chains